src/lock-fields-manager.ts

```diff
@@ -434,25 +446,43 @@ export class LockFieldsManager<U extends BaseMongoObject> {
 		return ret;
 	}
 
-	private pickOnlyNewValuesInArray(
+	private pickValuesInArray(
 		existingEntityArray: any[],
 		newEntityElement: any[],
 		currentPath: string,
+		pickOnlyNewValues: boolean,
 	): any[] {
 		const codeKeyNames = this._arrayWithReferences[currentPath];
 
 		const ret = [];
-		for (let i = 0; i < Math.max(existingEntityArray?.length, newEntityElement?.length); i += 1) {
+		for (let i = 0; i < Math.max(existingEntityArray.length, newEntityElement?.length); i += 1) {
```

Comment:
All branches are not coverred here : https://app.codecov.io/gh/neo9/n9-mongo-client/commit/5036c5e502ebbf64eab48392b80da234d31167f3#133bdafc660af83cc76ec06be078cb6a-R458

Reply:
as discussed, no tests will be added for this case in this PR